### PR TITLE
[CROW-88] Add Claim functionality

### PR DESCRIPTION
**Pull request by FacuEM**
#### :page_facing_up: Description:

In this PR we create a custom hook called useClaim that is in charge of interacting with the claim function of the contract.

---

#### :pushpin: Notes:

* The log shown in the console corresponds to the listener of the Claim event of the contract.

---

#### :play_or_pause_button: Demo:
https://www.loom.com/share/f07117e3f1fb49d69e6bff3cf3613498

### CODE REVIEW

File: pages/pledge/[id].page.js [hunk 1]
```diff
@@ -94,9 +101,13 @@ const PledgePage = () => {
             <Text>Revenue: ${campaign.fiatAmount}</Text>
             <Text>Pledged: ${campaign.currentAmount[0].amount}</Text>
 
-            <Button onClick={() => setIsModalOpen((prev) => !prev)}>
-              Pledge Now
-            </Button>
+            {isOwner ? (
```
Comment (marcosmartinez7) [inline pages/pledge/[id].page.js:104]: I think we need another condition that is: the goal is reached and the time ended 

but maybe you want to include that on another pr

File: utils/fetch.js [hunk 2]
```diff
@@ -6,7 +6,7 @@ export const fetchCampaigns = async () => {
 
   try {
     const { data } = await axios.get(
-      `${process.env.NEXT_PUBLIC_CROWDFUNDING_API}/campaigns?page=0&size=20`,
+      `${process.env.NEXT_PUBLIC_CROWDFUNDING_API}/campaigns?page=0&size=20&search=`,
```
Comment (marcosmartinez7) [inline utils/fetch.js:9]: Sorry but why is `search=` needed? dont understand if a parameter is not passed

File: pages/pledge/[id].page.js [hunk 3]
```diff
@@ -94,9 +101,13 @@ const PledgePage = () => {
...
+              <Button onClick={() => write?.()}>Claim</Button>
```
Comment (Tiago-Zarzavidjian) [inline pages/pledge/[id].page.js:105]: What's the explanation for using `?.()`?

File: utils/fetch.js (hunk 2 above)
Comment (FacuEM) [reply utils/fetch.js:9]: This is a BE bug that is fixed in an open PR, I'm going to delete it, it shouldn't be that &search="" I will delete it.

File: pages/pledge/[id].page.js (hunk 1 above)
Comment (FacuEM) [reply pages/pledge/[id].page.js:104]: Yes, I was thinking of adding it to another PR. This one does not take into account the UX of the claim, only the connection with the contract.

File: pages/pledge/[id].page.js (hunk 3 above)
Comment (FacuEM) [reply pages/pledge/[id].page.js:105]: Here is [Wagmi's documentation](https://wagmi.sh/react/hooks/useContractWrite) on how to implement a button that executes the write function.

The reason is that write can become null if there is a problem in the connection with the contract. For example: if a requirement of the contract is that the time in which it is expected to collect funds for a campaign has been fulfilled and it has not been fulfilled, write will be null.

Comment (FacuEM) [reply pages/pledge/[id].page.js:105]: I'll add the disabled prop

**Review by marcosmartinez7 — APPROVED** [review]

**Pull request merged**